ReChord_frontend.py

```diff
@@ -92,14 +95,27 @@ def search_snippet(path, snippet):
     Return: rendered result page 'ReChord_result.html'
     """
     xml = BytesIO(snippet.encode())
-    input_xml_tree = etree.parse(xml)  # pylint: disable=c-extension-no-member
-
-    named_tuples_ls = snippet_search_folder(path, input_xml_tree)
-
-    return render_template('ReChord_result.html', origins=named_tuples_ls)
+    try:
+        input_xml_tree, _ = prepare_tree(xml)  # pylint: disable=c-extension-no-member
+
+        named_tuples_ls = snippet_search_folder(path, input_xml_tree)
+        total_appearance = len(named_tuples_ls)
+        if total_appearance != 0:
+            return render_template('ReChord_result.html', origins=named_tuples_ls)
+        else:
+            not_found = "No matched snippet found, maybe try something else?"
+            return  render_template('ReChord_result.html', nomatch=not_found)
+    except:
+        if etree.XMLSyntaxError or ValueError:
+            error_msg = "Invalid MEI snippet inputs. Please double check the source and try it again!"
+        elif KeyError:
+            error_msg = "Invalid upload file. Please double check the source and try it again!"
+        else:
+            error_msg = "There are some errors in the search. Please check the source."
+
+        return render_template('ReChord_result.html', errors=error_msg)
```

Comment:
The "No exception type(s) specified" lint message means you need to specify the exception. Try:

```python
except (etree.XMLSyntaxError, ValueError):
    error_msg = "Invalid MEI snippet inputs. Please double check the source and try it again!"
except KeyError:
    error_msg = "Invalid upload file. Please double check the source and try it again!"
return render_template('ReChord_result.html', errors=error_msg)
```